scraping/scraper.py

```diff
@@ -127,57 +133,83 @@ def scrape_job_run(workspace, job_run_dict, session, result):
         run_type=job_run_dict["run_type"]
     )
+
+    if "creator_user_name" in job_run_dict:
+        job_run.creator_user_name = job_run_dict.get("creator_user_name")
+
     session.merge(job_run)
     result.num_job_runs += 1
 
 
+def get_task_type(settings):
+    task_types = ['notebook_task', 'spark_jar_task',
+                  'spark_python_task', 'spark_submit_task']
+    for task_type in task_types:
+        if settings.get(task_type) is not None:
+            return task_type.upper()
+
+    log.warning("Couldn't determine job task type, falling back to 'UNKNOWN'.")
+    return 'UNKNOWN'
+
+
 def scrape_jobs(workspace, job_dict, session, api, result):
     log.debug(f"Scraping job, id: {job_dict['job_id']}")
+    settings = job_dict.get("settings", {})
     job = Job(
         job_id=job_dict["job_id"],
         created_time=to_time(job_dict["created_time"]),
-        creator_user_name=job_dict["creator_user_name"],
-        name=job_dict["settings"]["name"],
+        name=settings["name"],
         workspace_id=workspace.id,
-        max_concurrent_runs=job_dict["settings"]["max_concurrent_runs"],
-        timeout_seconds=job_dict["settings"]["timeout_seconds"],
-        email_notifications=job_dict["settings"]["email_notifications"],
-        new_cluster=job_dict["settings"]["new_cluster"],
-        schedule_quartz_cron_expression=job_dict["settings"].get(
-            "schedule", {}).get("quartz_cron_expression", None),
-        schedule_timezone_id=job_dict["settings"].get(
-            "schedule", {}).get("timezone_id", None),
-        task_type="NOTEBOOK_TASK",
-        notebook_path=job_dict["settings"].get(
-            "notebook_task", {}).get("notebook_path", None),
-        notebook_revision_timestamp=job_dict["settings"].get(
-            "notebook_task", {}).get("revision_timestamp", None),
+        max_concurrent_runs=settings["max_concurrent_runs"],
+        timeout_seconds=settings["timeout_seconds"],
+        email_notifications=settings["email_notifications"],
+        # TODO: determine how should we handle the diff between new/existing
+        # clusters
+        new_cluster=(settings
+                     .get("new_cluster",
+                          {"cluster_id": settings.get("existing_cluster_id")})),
+        schedule_quartz_cron_expression=(settings
+                                         .get("schedule", {})
+                                         .get("quartz_cron_expression")),
+        schedule_timezone_id=settings.get("schedule", {}).get("timezone_id"),
+        task_type=get_task_type(settings),
+        notebook_path=settings.get("notebook_task", {}).get("notebook_path"),
+        notebook_revision_timestamp=(settings
+                                     .get("notebook_task", {})
+                                     .get("revision_timestamp")),
     )
+
+    if "creator_user_name" in job_dict:
+        job.creator_user_name = job_dict.get("creator_user_name")
```

Comment:
See my other comments above on this one - it could be unnecessary but a default could be required.

Reply:
`session.merge()` is an upsert function so if we have the user who was deleted we don't want to lose that information. That wss the reason behind the removal of the overwrite with "DELETED" string.